generate/genhcl/genhcl.go

```diff
@@ -181,66 +184,76 @@ func Load(
 		}
 
 		if !condition {
-			logger.Trace().Msg("condition=false, block wont be evaluated")
-
 			hcls = append(hcls, HCL{
 				label:     name,
-				origin:    loadedHCL.origin,
+				origin:    origin,
 				condition: condition,
 			})
 
 			continue
 		}
 
-		logger.Trace().Msg("evaluating block")
+		asserts := make([]config.Assert, len(hclBlock.Asserts))
+		assertsErrs := errors.L()
+		assertFailed := false
+
+		for i, assertCfg := range hclBlock.Asserts {
+			assert, err := config.EvalAssert(evalctx.Context, assertCfg)
+			if err != nil {
+				assertsErrs.Append(err)
+				continue
+			}
+			asserts[i] = assert
+			if !assert.Assertion {
+				// TODO(Katcipis): check warning
+				assertFailed = true
+			}
+		}
+
+		if err := assertsErrs.AsError(); err != nil {
+			return nil, err
+		}
+
+		if assertFailed {
+			hcls = append(hcls, HCL{
+				label:     name,
+				origin:    origin,
+				condition: condition,
+				asserts:   asserts,
+			})
+			continue
+		}
 
 		gen := hclwrite.NewEmptyFile()
-		if err := copyBody(gen.Body(), loadedHCL.block.Body, evalctx); err != nil {
+		if err := copyBody(gen.Body(), hclBlock.Content.Body, evalctx); err != nil {
 			return nil, errors.E(ErrContentEval, sm, err,
 				"generate_hcl %q", name,
 			)
 		}
 
-		// TODO(i4k): filename in line below must be an absolute host path.
-		formatted, err := fmt.FormatMultiline(string(gen.Bytes()), loadedHCL.origin.String())
+		formatted, err := fmt.FormatMultiline(string(gen.Bytes()), hclBlock.Origin)
 		if err != nil {
-			// genhcl must always generate valid code that is formatable
-			// this is a severe internal error
-			logger.Error().
-				Err(err).
-				Stringer("origin", loadedHCL.origin).
-				Str("code", string(gen.Bytes())).
-				Str("label", name).
-				Msg("internal error formatting generated code")
-
 			panic(errors.E(sm, err,
-				"internal error: formatting generated code for generate_hcl %q", name,
+				"internal error: formatting generated code for generate_hcl %q:%s", name, string(gen.Bytes()),
 			))
 		}
 		hcls = append(hcls, HCL{
 			label:     name,
-			origin:    loadedHCL.origin,
+			origin:    origin,
 			body:      formatted,
 			condition: condition,
+			asserts:   asserts,
 		})
 	}
 
-	sort.Slice(hcls, func(i, j int) bool {
-		return hcls[i].String() < hcls[j].String()
+	sort.SliceStable(hcls, func(i, j int) bool {
```

Comment:
Changed to just using the label, I think we started using String when it was simpler. Currently it would also use condition/body/etc on the ordering and that seemed odd. I think just ordering by label ir organized enough for testing etc. Since labels can be repeated I used a stable sort to keep the origina order things are processed.